src/framework/scene-registry.js

```diff
@@ -347,6 +358,55 @@ class SceneRegistry {
         });
     }
 
+    /**
+     * Change to a new scene. Calling this function will load the scene data, delete all
+     * entities and graph nodes under `app.root` and load the scene settings and hierarchy.
+     *
+     * @param {SceneRegistryItem | string} sceneItem - The scene item (which can be found with
+     * {@link SceneRegistry#find} or name of the scene.".
+     * @param {ChangeSceneCallback} [callback] - The function to call after loading,
+     * passed (err, entity) where err is null if no errors occurred.
+     * @example
+     * app.scenes.ChangeScene("Scene Name", function (err, entity) {
+     *     if (!err) {
+     *         // success
+     *     } else {
+     *         // error
+     *     }
+     * });
+     */
+    changeScene(sceneItem, callback) {
+        if (!(sceneItem instanceof SceneRegistryItem)) {
+            sceneItem = this.find(sceneItem);
+        }
+
+        const self = this;
+
+        this._loadSceneData(sceneItem, false, function (err, sceneItem) {
+            if (err) {
+                if (callback) {
+                    callback(err);
+                }
+
+                return;
+            }
+
+            // Destroy/Remove all nodes on the app.root
+            const rootChildren = self._app.root.children;
+            while (rootChildren.length > 0) {
+                const child = rootChildren[0];
+                if (child instanceof Entity) {
+                    child.destroy();
+                } else {
+                    child.parent.removeChild(child);
+                }
+            }
```

Comment:
Is there a nicer way to do this? This feels pretty bad?

Reply:
this should work
```
while (rootChildren.length > 0) {
    const child = rootChildren[0];
    child.reparent(null);
    child.destroy?.();
}
```